astropy/time/core.py

```diff
@@ -1314,10 +1314,11 @@ def scale(self, val):
     def _check_val_type(self, val1, val2):
         """Input value validation, typically overridden by derived classes"""
         try:
-            assert (val1.dtype == np.double and
-                    (val2 is None or val2.dtype == np.double))
+            assert (val1.dtype == np.double and np.all(np.isfinite(val1)) and
+                    (val2 is None or
+                     val2.dtype == np.double and np.all(np.isfinite(val2))))
         except:
```

Comment:
Should this be `except AssertionError`?  `np.all` or `np.isfinite` could theoretically raise different exceptions, and we wouldn't want to swallow those up.